From: Ilya Dryomov <idryomov@gmail.com>
Date: Fri, 22 Jun 2012 12:14:13 -0600
Patch-mainline: yes
References: FATE#306586
Git-commit: 44c44af2f4a6dc1595f1711cf307bd01062fd129
Subject: [PATCH] Btrfs: do not ignore errors from
 btrfs_cleanup_fs_roots() when mounting

There used to be a BUG_ON(ret) there before EH patch (79787eaa) went in.
Bail out with EINVAL.

Cc: David Sterba <dsterba@suse.cz>
Signed-off-by: Ilya Dryomov <idryomov@gmail.com>
Signed-off-by: David Sterba <dsterba@suse.cz>
---
 fs/btrfs/disk-io.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/fs/btrfs/disk-io.c b/fs/btrfs/disk-io.c
index fd216d9..dd6676b 100644
--- a/fs/btrfs/disk-io.c
+++ b/fs/btrfs/disk-io.c
@@ -2466,8 +2466,8 @@ retry_root_backup:
 
 	if (!(sb->s_flags & MS_RDONLY)) {
 		ret = btrfs_cleanup_fs_roots(fs_info);
-		if (ret) {
-			}
+		if (ret)
+			goto fail_trans_kthread;
 
 		ret = btrfs_recover_relocation(tree_root);
 		if (ret < 0) {
-- 
1.7.6.233.gd79bc

